kyuubi-main/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala

```diff
@@ -27,16 +27,26 @@ import org.apache.hadoop.security.UserGroupInformation
 import org.apache.kyuubi._
 import org.apache.kyuubi.config.KyuubiConf
 import org.apache.kyuubi.config.KyuubiConf.ENGINE_SPARK_MAIN_RESOURCE
+import org.apache.kyuubi.config.internal.Tests
 import org.apache.kyuubi.engine.ProcBuilder
 
 class SparkProcessBuilder(
     override val proxyUser: String,
     override val conf: KyuubiConf,
-    override val env: Map[String, String] = sys.env)
+    envMap: Map[String, String] = sys.env)
   extends ProcBuilder with Logging {
 
   import SparkProcessBuilder._
 
+  override protected def env: Map[String, String] = {
+    val testingHadoopConfDir = conf.getOption(Tests.TESTING_HADOOP_CONF_DIR.key)
+    if (Utils.isTesting && testingHadoopConfDir.isDefined) {
+      envMap ++ Map("HADOOP_CONF_DIR" -> testingHadoopConfDir.get)
```

Comment:
can we set HADOOP_CONF_DIR in pom.xml to a dummy dir like `project.build`?

Reply:
I remembered I tried it before, let me try again.

Reply:
It does not work, it can not be transferred to process launched by command.

Reply:
how about we pass environments to `SparkProcessBuilder`, as it's designed to support this

Reply:
Can we create a separate PR first to make it work?

Reply:
sure, will create a separate pr to support transfer env map.